src/Sarif.Viewer.VisualStudio/CodeAnalysisResultManager.cs

```diff
@@ -239,24 +243,89 @@ public bool TryRebaselineCurrentSarifError(string uriBaseId, string originalFile
+                rebaselinedFile = CreateFileFromContents(originalFilename);
             }
             else
             {
-                rebaselinedFile = GetRebaselinedFileName(uriBaseId, originalFilename);
-            }
+                if (Uri.IsWellFormedUriString(originalFilename, UriKind.Absolute))
+                {
+                    // File needs to be downloaded.
+                    rebaselinedFile = DownloadFile(originalFilename);
+                }
+                else
+                {
+                    // User needs to locate file.
+                    rebaselinedFile = GetRebaselinedFileName(uriBaseId, originalFilename);
+                }
 
-            if (String.IsNullOrEmpty(rebaselinedFile) || originalFilename.Equals(rebaselinedFile, StringComparison.OrdinalIgnoreCase))
-            {
-                return false;
+                if (String.IsNullOrEmpty(rebaselinedFile) || originalFilename.Equals(rebaselinedFile, StringComparison.OrdinalIgnoreCase))
+                {
+                    return false;
+                }
             }
 
             CurrentSarifError.RemapFilePath(originalFilename, rebaselinedFile);
             return true;
         }
 
+        // Contents are embedded in SARIF. Create a file from these contents.
+        private string CreateFileFromContents(string fileName)
+        {
+            var fileData = FileDetails[fileName];
+
+            // Get temporary path.
+            string path = Path.GetTempPath();
+            path = Path.Combine(path, TemporaryFileDirectoryName);
+
+            // If the file path already starts with the temporary location,
+            // that means we've already built the temporary file, so we can
+            // just open it.
+            if (fileName.StartsWith(path))
+            {
+                path = fileName;
+            }
+            // Else we have to create a location under the temp path.
+            else
+            {
+                // If the file path is absolute, or if it is relative to an
+                // unknown root, we need to strip the drive letter and/or
+                // leading slash, so that we can relate the file path to an
+                // arbitrary remapped disk location.
+                string filePath = fileName;
+                filePath = filePath.Substring(Path.GetPathRoot(filePath).Length);
+
+                path = Path.Combine(path, fileData.Sha256Hash);
+                path = Path.Combine(path, filePath);
+            }
+
+            string directory = new FileInfo(path).Directory.FullName;
+
+            if (!Directory.Exists(directory))
+            {
+                Directory.CreateDirectory(directory);
+            }
+
+            if (!File.Exists(path))
+            {
+                string contents = fileData.GetContents();
+                File.WriteAllText(path, contents);
+                // File should be readonly, because it is embedded.
```

Comment:
>File should be readonly, because it is embedded [](start = 19, length = 47)

Is there also a way to mark the editing buffer as readonly? As the code stands, I can type in it, and Save As it somewhere else. Is that ok? Thoughts?